### PR TITLE
[Remove Blocking usages]LPS-193319 portal-kernel | WorkflowDefinitionManagerUtil

### PR DESCRIPTION
**Pull request by JanisZhang**
https://liferay.atlassian.net/browse/LPS-193319

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:workflow

**Comment by JanisZhang** [comment]
ci:test:startup

**Comment by Kvale1733** [comment]
ci:stop

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
ci:test:workflow

File: modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/display/context/KaleoFormsAdminDisplayContext.java [hunk 1]
```diff
@@ -401,12 +403,10 @@ _renderRequest, _getIteratorURL(), null,
 					emptyResultsMessage);
 
 			searchContainer.setResultsAndTotal(
-				() ->
-					WorkflowDefinitionManagerUtil.getActiveWorkflowDefinitions(
-						_themeDisplay.getCompanyId(),
-						searchContainer.getStart(), searchContainer.getEnd(),
-						null),
-				WorkflowDefinitionManagerUtil.getActiveWorkflowDefinitionsCount(
+				() -> _workflowDefinitionManager.getActiveWorkflowDefinitions(
```
Comment (dantewang) [inline modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/display/context/KaleoFormsAdminDisplayContext.java:406]: @JanisZhang The change to this class doesn't seem to be related to `KaleoFormsUtil`, please put it in a separate commit.

File: modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java [hunk 2]
```diff
@@ -267,6 +268,10 @@ private void _renderKaleoProcess(RenderRequest renderRequest)
 				WorkflowTaskManagerUtil.getWorkflowTask(
 					themeDisplay.getCompanyId(), workflowTaskId));
 		}
+
+		renderRequest.setAttribute(
```
Comment (dantewang) [inline modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java:272]: @JanisZhang I think you can add a method `KaleoFormsAdminDisplayContext.getWorkflowDefinitionManager` so that you don't need to add it to the request.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@GinsonRen @111Anna Please run startup diff for this pull, and also check all service tracker numbers, thanks!

File: modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java (hunk 2 above)
Comment (JanisZhang) [reply modules/dxp/apps/portal-workflow-kaleo-forms/portal-workflow-kaleo-forms-web/src/main/java/com/liferay/portal/workflow/kaleo/forms/web/internal/portlet/KaleoFormsAdminPortlet.java]: @dantewang force-pushed. thanks!

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:startup

**Comment by JanisZhang** [comment]
ci:test:workflow

**Comment by 111Anna** [comment]
Hi @dantewang @GinsonRen,  I ran the PR on CE and DXP, the test results are normal. 

**About the  service tracker numbers:**
Check the totals, there is no significant change, as shown in the table below.
Check in Back Tracks, there is a significant difference. The number of PR's service tracker compared to Baseline's has decreased by a total of 173.
- 1 from  ServiceWrapperRegistry
- 1 from  LogServiceEnabhttps://drive.google.com/drive/u/0/folders/1t0hgh6NBkp5R_hauD4PVOer9Irv7wN-9ledLogger
- 18 from  ServiceTrackerFactory
- 153  from  ContextController


CE:
Portal Version |GitId | Average Startup (ms) | % | Average Shutdown (ms) | % | Open File |Thread Create |Open Prepared Statement |Open SQL|Open service tracker|Open bundle tracker
-- | -- | -- | -- | -- | -- | -- | -- | -- | --|--|--
Baseline | 9c4968c53 |19164.6 | - | 2592.6 | - | 1480 | 95 | 120 | 855|5518|32
Baseline + PR-5242 | - | 19166.6 | -0.01% | 2594.6 | -0.07% |  1480|  95| 120 | 854|5517|33

DXP:
Portal Version | GitId | Average Startup (ms) | % | Average Shutdown (ms) | % | Open File |Thread Create |Open Prepared Statement |Open SQL|Open service tracker|Open bundle tracker
-- | -- | -- | -- | -- | -- | -- | -- | -- | --|--|--
Baseline | 9c4968c53 |20531.6| - | 3087.3| - | 1604 | 96 | 126 | 931|5962|33
Baseline + PR-5242 | - | 20604.3 | -0.35% |  3050.6| 1.12% | 1604 |97 | 126 |932 |5962|33

**Comment by Kvale1733** [comment]
@JanisZhang Regression found. Cannot create Dynamic Data Lists

Steps to reproduce
1. Go to Content & Data > Dynamic Data Lists
2. Click the plus button to add an entry

Expected Results: Can create a Dynamic Data List

Actual Results:
DDL

Also the following error appears in the console
```
[beanshell] 2023-08-14 07:22:06.407 ERROR [http-nio-8080-exec-8][PortletRequestDispatcherImpl:295] Unable to dispatch request: java.lang.NullPointerException
[beanshell] 2023-08-14 07:22:06.407 ERROR [http-nio-8080-exec-8][PortletServlet:109] Unable to process portlet com_liferay_dynamic_data_lists_web_portlet_DDLPortlet: org.apache.jasper.JasperException: java.lang.NullPointerException
[beanshell] javax.portlet.PortletException: org.apache.jasper.JasperException: java.lang.NullPointerException
[beanshell] 	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.dispatch(PortletRequestDispatcherImpl.java:298) ~[portal-impl.jar:?]
[beanshell] 	at com.liferay.portlet.internal.PortletRequestDispatcherImpl.include(PortletRequestDispatcherImpl.java:114) ~[portal-impl.jar:?]
[beanshell] 	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.include(MVCPortlet.java:611) ~[portal-kernel.jar:?]
[beanshell] 	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.include(MVCPortlet.java:627) ~[portal-kernel.jar:?]
[beanshell] 	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.doDispatch(MVCPortlet.java:497) ~[portal-kernel.jar:?]
[beanshell] 	at com.liferay.dynamic.data.lists.web.internal.portlet.DDLPortlet.doDispatch(DDLPortlet.java:146) ~[bundleFile:?]
[beanshell] 	at javax.portlet.GenericPortlet.render(GenericPortlet.java:291) ~[portlet.jar:3.0.1]
[beanshell] 	at com.liferay.portal.kernel.portlet.bridges.mvc.MVCPortlet.render(MVCPortlet.java:300) ~[portal-kernel.jar:?]
```
 [Log](https://testray.liferay.com/reports/production/logs/2023-08/test-1-27/test-portal-acceptance-pullrequest(master)/9525/functional-tomcat90-mysql57-jdk8/2/8/jenkins-console.txt.gz?authuser=0)
